lib/ex_unit/test/ex_unit/doc_test_test.exs

```diff
@@ -200,6 +200,62 @@ defmodule ExUnit.DocTestTest.IndentationNotEnough do
   def test_fun, do: :ok
 end |> write_beam
 
+defmodule ExUnit.DocTestTest.FencedHeredocs do
+  @doc ~S'''
+  Receives a test and formats its failure.
+
+  ## Examples
+
+  ```
+  iex> 1 + 2
+  3
+  ```
+
+      ```
+      iex> 1 + 2
+      3
+      ```
+
+  ```
+      iex> 1 + 2
+      3
+  ```
+  '''
+  def heredocs, do: :ok
+end |> write_beam
+
+defmodule ExUnit.DocTestTest.FencedIndentationMismatchedPrompt do
```

Comment:
They helped convinced me the change was benign, however, I'd meant to make the mismatched tests ensure it could handle fence indentation mismatches as well.

Addtionally, `ExUnit.DocTestTest.FenceIncomplete` should in fact pass, per CommonMark spec, however it was not because the exunit case inside it was incomplete. Fixing these things.